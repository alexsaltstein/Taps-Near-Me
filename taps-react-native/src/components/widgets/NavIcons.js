import React from 'react';
import { TouchableOpacity, Image, StyleSheet, Text } from 'react-native';
import { COLORS } from '../../styles/COLORS';

const NavIcons = ({ source, to, navigation }) => {
  return (
    <TouchableOpacity
      onPress={() => navigation.navigate(to)}
      style={styles.navButton}>
      <Image
        source={source}
        style={styles.navIcon} />
    </TouchableOpacity>
  )
};

const styles = StyleSheet.create({
  navButton: {
    paddingHorizontal: 8,
  },
  navIcon: {
    width: 40,
    height: 40,
  }
});

export default NavIcons;