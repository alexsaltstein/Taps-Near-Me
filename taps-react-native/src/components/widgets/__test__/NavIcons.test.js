import '../../../matchMedia.mock';
import React from 'react';
import renderer, { act } from 'react-test-renderer';
import NavIcons from '../NavIcons';
import { render, fireEvent } from '@testing-library/react-native'
import ReactDOM from 'react-dom';

const mockNavigation = {
    navigate: jest.fn(),
    addListener: jest.fn()
}

test('renders correctly', () => {
    const tree = renderer.create(<NavIcons/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('navigates to screens correctly', () => {
    const tree = renderer.create(<NavIcons source={'filter'} to={'settings'} navigation={ mockNavigation } />);
    const root = tree.root;
    const prop = root.findByProps({testID:'NavIcons.Button'});
    prop.props.onPress();
    expect(mockNavigation.navigate).toBeCalledWith('settings');
})