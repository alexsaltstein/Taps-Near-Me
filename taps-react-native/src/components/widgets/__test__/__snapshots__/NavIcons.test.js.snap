// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n r-cursor-1loqt21 r-paddingHorizontal-1pn2ns4 r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
  data-focusable={true}
  data-testid="NavIcons.Button"
  onClick={[Function]}
  onContextMenu={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "WebkitTransitionDuration": "0s",
      "transitionDuration": "0s",
    }
  }
  tabIndex="0"
>
  <div
    className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
    style={
      Object {
        "height": "40px",
        "width": "40px",
      }
    }
  >
    <div
      className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
    />
  </div>
</div>
`;
